File: CASHONEWebsiteNET5/Models/CashoneCart/CartItem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace Application.Models.CashoneCart
{
    [Serializable]
    public class CartItem : IComparable
    {
        [Key]
        public Guid ItemId { get; set; }
        public string ItemName { get; set; }
        public string ItemCode { get; set; }
        public short AvailableQuantity { get; set; }
        public short Quantity { get; set; }
        public decimal Cost { get; set; }
        public decimal? Discount { get; set; }

        public int CompareTo(object obj)
        {
            return ItemName.CompareTo(((CartItem)obj).ItemName);
        }
        public override int GetHashCode()
        {
            return ItemId.GetHashCode();
        }
    }
}

File: CASHONEWebsiteNET5/wwwroot/model/CartItem.js
function CartItem(data) {
    var instance = this;

    instance._datakey = data.ItemId;

    instance.Uid = ko.observable(data.ItemId);
    instance.OrderId = ko.observable(data.OrderId);
    instance.ItemId = ko.observable(data.ItemId);
    instance.ItemName = ko.observable(data.ItemName);
    instance.ItemCode = ko.observable(data.ItemCode);
    instance.AvailableQuantity = ko.observable(data.AvailableQuantity);
    instance.Quantity = ko.observable(data.Quantity);
    instance.Cost = ko.observable(data.Cost);
    instance.Discount = ko.observable((data.Discount === null || data.Discount === undefined) ? 0 : data.Discount);

    instance.PicturePath = ko.computed(function () {
        return '/home/getpicture/' + instance.ItemCode();
    });

    instance.LineTotal = ko.computed(function () {
        return (instance.Quantity() * instance.Cost()) - (instance.Quantity() * instance.Discount());
    });
}